compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -18897,10 +18897,17 @@ private STNode getTypeDescFromExpr(STNode expression) {
         }
     }
 
-    private List<STNode> getBindingPatternsList(List<STNode> ambibuousList) {
+    private List<STNode> getBindingPatternsList(List<STNode> ambibuousList, boolean isListBP) {
         List<STNode> bindingPatterns = new ArrayList<>();
         for (STNode item : ambibuousList) {
-            bindingPatterns.add(getBindingPattern(item));
```

Comment:
Shall we bring `isListBP` flag inside the `getBindingPattern()`? there you can check the flag = true in `QUALIFIED_NAME_REFERENCE` case and if so `break` and then execute the error reporting we do under the default case.

Reply:
I think the error given at the default case is okay to use for this as well.